### ISSUE TITLE
[Labour#1]: Port wimp to use hercules

### ISSUE THREAD

**Issue by kshitij10496** [kshitij10496/hercules#17]
@themousepotato Can you experiment with using `hercules` API to power `wimp`?
Given that the endpoints currently have bare minimum documentation, I would be happy to assist you. 😄 

cc @icyflame @amrav

**Comment by themousepotato** [comment]
I've already opened an issue https://github.com/metakgp/wimp/issues/39 4 days ago and was waiting for someone to take it. :D I can't think of myself doing that. It'll delete most of my code :(

**Comment by kshitij10496** [comment]
> I can't think of myself doing that. It'll delete most of my code :(

Yes, I agree with the fact that most of the scrapping logic in `wimp` would become redundant. 😢 
Having said that, I believe that `hercules` would make the project extensible and take a lot of load off of your shoulders. Moreover, the integration would help develop a better understanding of dealing with JSON APIs in Python, specifically related to making requests and decoding responses. 😅

**Comment by themousepotato** [comment]
Will definitely work on this sometime, then @kshitij10496  :)

**Comment by icyflame** [comment]
@themousepotato Getting to delete old code and write new, better code is always a privilege! In these projects, we have the time and resources to re-write stuff well, that's a great opportunity. I often learn a lot about the mistakes I made the first time around whenever I do a re-write of something I made (eg: bar-horizontal and cli-cube-timer, I wrote both before I knew what Node was really for, and after doing Node for a couple years, I re-wrote them! Couldn't believe the kind of Node I used to write!) If you get the chance to, you should definitely be the one to re-write wimp!

**Comment by themousepotato** [comment]
Sure @icyflame :D
